src/Thenewboston/Bank/Api/InvalidBlocksService.cs

```diff
@@ -18,7 +18,7 @@ public InvalidBlocksService(IHttpRequestSender requestSender)
             _requestSender = requestSender;
         }
         
-        public async Task<IEnumerable<BankInvalidBlock>> GetInvalidBankBlocks()
+        public async Task<IEnumerable<BankInvalidBlock>> GetInvalidBankBlocksAsync()
         {
             try
             {
```

Comment:
This method should return a PaginatedResponseModel instead of IEnumerable<T>. The documentation is a bit out of date, and this has been changed since it was written. We are also working to have that updated.

Reply:
Hello @joshman1019 please which query parameter handles the pagination for the API here??

Reply:
I did a similar thing with the Confirmation Blocks. You'll find the service here: https://github.com/thenewboston-developers/dotnetcore-sdk/blob/develop/src/Thenewboston/Bank/Api/BankConfirmationBlockService.cs

When the bank or validator serializes the data on that end, it will send over an abject that matches those parameters. And so when you create your test, you can mock up the "received data" using your BankInvalidBlock model. Take a look at my test for Confirmation Blocks to get you started: https://github.com/thenewboston-developers/dotnetcore-sdk/blob/develop/src/Thenewboston.Tests/Bank/Api/BankConfirmationBlockServiceTests.cs

To simplify it, your service will de-serialize the string version of the PaginatedResponseModel sent over from the API, and the "Results" property of that model will contain a list of your BankInvalidBlocks. Eventually we will create the necessary features to convert those into BankInvalidBlock objects, but for now the PaginatedResponseModel treats it as a plain object.  Hopefully those two references will help to guide you along. The reason they are using this model is most likely to save resources since only a limited number of results are returned at one time. This also makes paging through results a lot easier on the client side rather than just receiving a huge dump all at once. Developers using this SDK will probably also appreciate that they will not have to write that code... I know that would be the case for me haha.

Reply:
Oh, also. You can take a look at what a live bank is returning by visiting this address: http://143.110.137.54/invalid_blocks 

Unfortunately...and probably fortunately (lol)... there are no results, but this is sometimes a good way to see how the live servers are sending over data. Unfortunately it is difficult with POST or PATCH requests, but for GET it is really helpful.

Reply:
And... one more comment...sorry. The Next and Previous properties of the PaginatedResponseModel will contain the API calls necessary to get the next or previous set of results.

Reply:
II looked ad the Django Rest Framework Doc to implement a PaginationParams for The page scernario and limit with offset scenario too